js/data/array_utils.js

```diff
@@ -78,21 +79,50 @@ function setDataByKeyMapValue(array, key, data) {
     }
 }
 
-function applyBatch(keyInfo, array, batchData, groupCount, useInsertIndex) {
+function createObjectWithChanges(target, changes) {
+    const result = target ? Object.create(Object.getPrototypeOf(target)) : {};
+    const targetWithoutPrototype = extendFromObject({}, target);
+
+    objectUtils.deepExtendArraySafe(result, targetWithoutPrototype, true, true);
+    return objectUtils.deepExtendArraySafe(result, changes, true, true);
+}
+
+function applyBatch({ keyInfo, array, batchData, groupCount, useInsertIndex, immutable, enableCache }) {
```

Comment:
Rename array to data, batchData to changes to be consistent with applyChanges.

Reply:
ok